File: map_viewer.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

void load_and_display_map(const std::string& file_path) {
    std::ifstream map_file(file_path);
    if (!map_file.is_open()) {
        std::cerr << "Error: Could not open file " << file_path << std::endl;
        return;
    }

    std::string line;
    while (std::getline(map_file, line)) {
        std::cout << line << std::endl;
    }

    map_file.close();
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cout << "Usage: ./map_viewer <path_to_map_file>" << std::endl;
        return 1;
    }

    load_and_display_map(argv[1]);

    return 0;
}

File: map_viewer.py
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage: python map_viewer.py <path_to_map_file>")
        sys.exit(1)

    map_file_path = sys.argv[1]
    load_map(map_file_path)

def load_map(file_path):
    try:
        with open(file_path, 'r') as f:
            map_data = f.read()
        print(map_data)
    except FileNotFoundError:
        print(f"Error: The file '{file_path}' was not found.")
        sys.exit(1)

if __name__ == "__main__":
    main()
